skimage/transform/_geometric.py

```diff
@@ -224,8 +241,81 @@ def residuals(self, src, dst):
         """
         return np.sqrt(np.sum((self(src) - dst) ** 2, axis=1))
 
+    @classmethod
+    def identity(cls, dimensionality=None):
+        """Identity transform
+
+        Parameters
+        ----------
+        dimensionality : {None, 2}, optional
+            This transform only allows dimensionality of 2, where None
+            corresponds to 2. The parameter exists for compatibility with other
+            transforms.
+
+        Returns
+        -------
+        tform : transform
+            Transform such that ``np.all(tform(pts) == pts)``.
+        """
+        return cls(dimensionality=dimensionality)
```

Comment:
I remember that we talked about this but that there was some confusion. So for clarification: do we want to get rid of the `dimensionality=` argument eventually?

If so why not make this method construct the identity matrix and pass it to `_HMatrixTransform(matrix=...)`, instead of constructing it in its init? 

Also shouldn't this currently conflict with inheriting classes such as `PiecewiseAffineTransform`?

Reply:
@matthew-brett, pinging you in case this got lost in the overall discussion.

Reply:
I still don't understand the merit in keeping `dimensionality=` around in the `__init__`.

Reply:
> I still don't understand the merit in keeping `dimensionality=` around in the `__init__`.

It's just for back-compatibility.   In due course we can deprecate and remove `dimensionality` in the `__init__`.

Reply:
> If so why not make this method construct the identity matrix and pass it to `_HMatrixTransform(matrix=...)`, instead of constructing it in its init?

The only problem comes where the inheritor is not a subclass of `_HMatrixTransform`.    Actually, that code is never called, so I now raise a `NotImplementedError`.

Yes, `PiecewiseAffineTransform` is obliged to (and does) override the `__init__`.